src/jit/lsra.cpp

```diff
@@ -6653,26 +6653,24 @@ LinearScan::allocateRegisters()
                 }
                 else
 #endif // FEATURE_SIMD
-                if (currentRefPosition->IsActualRef())
+                if (currentRefPosition->RequiresRegister())
```

Comment:
This should be either `if (currentRefPosition->RequiresRegister() || currentRefPosition->AllocateIfProfitable())`, OR in the above, you should define something like a `bool considerSpilling`, which is only set to true under these two conditions.